src/instructlab/model/backends/backends.py

```diff
@@ -29,3 +32,107 @@ def run(self):
     @abc.abstractmethod
     def shutdown(self):
         """Shutdown serving backend"""
+
+
+def is_model_gguf(model_path: pathlib.Path) -> bool:
+    """
+    Check if the file is a GGUF file.
+    Args:
+        model_path (Path): The path to the file.
+    Returns:
+        bool: True if the file is a GGUF file, False otherwise.
+    """
+    # Standard
+    import mmap
+    import struct
+
+    # Third Party
+    from gguf.constants import GGUF_MAGIC
+
+    with open(model_path, "rb") as f:
+        # Memory-map the file on the first 4 bytes (this is where the magic number is)
+        mmapped_file = mmap.mmap(f.fileno(), length=4, access=mmap.ACCESS_READ)
+
+        # Read the first 4 bytes
+        first_four_bytes = mmapped_file.read(4)
+
+        # Convert the first four bytes to an integer
+        first_four_bytes_int = struct.unpack("<I", first_four_bytes)[0]
+
+        # Close the memory-mapped file
+        mmapped_file.close()
+
+        return first_four_bytes_int == GGUF_MAGIC
```

Comment:
Why do you use mmap here? I kinda get it. `mmap` is a great way to deal with large files and too few people know about the power of mmap. It seems a bit overpowered here, though. A simple `read()` would do the trick, too. Also it's a good idea to close the file as early as possible and do the rest of the work outside the context manager.

```suggestion
    with open(model_path, "rb") as f:
        first_four_bytes = f.read(4)

    # Unpack first four bytes as little endian uint32
    first_four_bytes_int = struct.unpack("<I", first_four_bytes)[0]

    return first_four_bytes_int == GGUF_MAGIC
```

Reply:
Absolutely, I'm using it for efficiency. It might be a bit overpowered, but at least it helps more people become aware of it 😅. Anyway, I'll just proceed with a simple read.